src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -121,60 +119,56 @@
     }
 }
 
-static Parser CreateCommandLineParser()
+static CliRootCommand CreateCommandLineParser()
 {
-    var debugOption = new Option<bool>("--debug", getDefaultValue: () => false)
+    var debugOption = new CliOption<bool>("--debug")
     {
         Description = "Flag indicating if the debugger should be launched on startup.",
-        IsRequired = false,
+        Required = false,
+        DefaultValueFactory = _ => false,
     };
-    var brokeredServicePipeNameOption = new Option<string?>("--brokeredServicePipeName")
+    var brokeredServicePipeNameOption = new CliOption<string?>("--brokeredServicePipeName")
     {
         Description = "The name of the pipe used to connect to a remote process (if one exists).",
-        IsRequired = false,
+        Required = false,
     };
 
-    var logLevelOption = new Option<LogLevel>("--logLevel", description: "The minimum log verbosity.", parseArgument: result =>
+    var logLevelOption = new CliOption<LogLevel>("--logLevel")
     {
-        var value = result.Tokens.Single().Value;
-        return !Enum.TryParse<LogLevel>(value, out var logLevel)
```

Comment:
@dibarbet Were enums just not supported on the version we were on or was this necessary for some special reason?

Reply:
I honestly don't remember.  If this works with VSCode then I'm good with it.

Reply:
enums are now supported OOTB, I can not speak about the past as I've been involved in this project only for a year and a half